### PR TITLE
Add SizeCredit

### PR DESCRIPTION
**Pull request by aviggiano**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### List of audit links if any:

- https://github.com/spearbit/portfolio/blob/master/pdfs/Size-Spearbit-Security-Review.pdf
- https://github.com/solidified-platform/audits/blob/master/Audit%20Report%20-%20Size%20Lending.pdf

##### Website Link:

- https://size.credit/

##### Logo (High resolution, will be shown with rounded borders):

- https://uploads-ssl.webflow.com/666ebfc20f3c32e295f62dc5/6696d2366ce07eaaaaecd2b7_Size%20Logo.svg

##### Current TVL:

- $1.35 k

##### Treasury Addresses (if the protocol has treasury)

- N/A

##### Chain:

- Base

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)

- N/A

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)

- N/A

##### Short Description (to be shown on DefiLlama):

- Size is a credit marketplace with unified liquidity across maturities.

##### Token address and ticker if any:

- N/A

##### Category (full list at https://defillama.com/categories) *Please choose only one:

- Lending

##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):

- Chainlink

##### Implementation Details: Briefly describe how the oracle is integrated into your project:

- Chainlink oracles are used to calculate the price ETH/USD and USDC/USD determine the collateral token price in terms of borrow token price

##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:

- https://docs.size.credit/

##### forkedFrom (Does your project originate from another project):

- N/A

##### methodology (what is being counted as tvl, how is tvl being calculated):

- WETH (collateral token) + USDC deposited on Aave v3 (borrow token)

##### Github org/user (Optional, if your code is open source, we can track activity):

- https://github.com/SizeCredit

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/size-credit exports TVL: 
        
 
 
    base                      1.35 k
    base-borrowed             0
    borrowed                  0
    
    total                    1.35 k

**Comment by llamatester** [comment]
The adapter at projects/size-credit exports TVL: 
        
 
 
    base                      1.35 k
    base-borrowed             0
    borrowed                  0
    
    total                    1.35 k

File: projects/size-credit/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const SIZE_PROXY_CONTRACT = '0xC2a429681CAd7C1ce36442fbf7A4a68B11eFF940'
+const tokens = [
+  ADDRESSES.base.WETH,
+  AUSDC_CONTRACT,
+]
+
+async function borrowed(api) {
+  const totalDebt = await api.call({ abi: 'erc20:totalSupply', target: SZDEBT_CONTRACT, });
+
+  return api.add(ADDRESSES.base.USDbC, totalDebt)
```
Comment (g1nt0ki) [inline projects/size-credit/index.js:15]: @aviggiano thanks for the PR, this is the part I am unsure of, we dont price SZDEBT_CONTRACT token, but its value will be fixed to 1 USDC?

Comment (aviggiano) [reply projects/size-credit/index.js:15]: Hello

Thank you for your time.

Indeed the debt token represents a USDC value, so your fix is correct.

**Pull request merged**